### PR TITLE
Friend system

### PR DESCRIPTION
**Pull request by AttiaYoussef**
# Added

- Friend system on the database
- Friend screen in settings so user can add/see their friends
- Possibility to add a friend directly from the room
- When inviting someone, give the option to scroll through friends to add them without typing their name explicitly.

# TO DO

- [x] Finish implementing the features
- [x] Fix tests 
- [x] Have a good enough coverage

### CODE REVIEW

File: app/src/main/java/io/github/polysmee/appointments/fragments/AppointmentCreationAddUserFragment.java [hunk 1]
```diff
@@ -148,6 +160,53 @@ public void reset() {
...
+               if(!invites.contains(s)){
+                   invites.add(s);
+                   dataPasser.dataPass(invites, AppointmentActivity.INVITES);
+                   searchInvite.setText("");
+
+                   if (mode == AppointmentActivity.DETAIL_MODE) {
+                       removedInvites.remove(s);
+                       dataPasser.dataPass(removedInvites, AppointmentActivity.REMOVED_INVITES);
+                   }
+
```
Comment (MrZaiko) [inline app/src/main/java/io/github/polysmee/appointments/fragments/AppointmentCreationAddUserFragment.java]: A function could make it clearer

File: app/src/main/java/io/github/polysmee/room/fragments/RoomActivityParticipantsFragment.java [hunk 2]
```diff
@@ -167,6 +169,16 @@ public void onDone(long o) {
...
+                    MainUserSingleton.getInstance().getFriends_Once_And_Then((friendsIds)->{
+                        if(friendsIds.contains(id)){
+                            friendshipButton.setImageResource(R.drawable.baseline_remove);
+                        }
+                        else{
+                            friendshipButton.setImageResource(R.drawable.baseline_add);
+                        }
+                    });
+                    friendshipButton.setVisibility(View.VISIBLE);
+                    friendshipButton.setOnClickListener((v)->{friendshipButtonBehavior(v,id);});
```
Comment (MrZaiko) [inline app/src/main/java/io/github/polysmee/room/fragments/RoomActivityParticipantsFragment.java:181]: Maybe add a test for this button

File: app/src/main/res/drawable/baseline_remove.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
```
Comment (MrZaiko) [inline app/src/main/res/drawable/baseline_remove.xml]: 36dp

**Review by MrZaiko — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/github/polysmee/settings/FriendsActivity.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    }
+
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        user.removeFriendsListener(friendsValuesListener);
+    }
+
+    protected void attributeSet(){
```
Comment (Morpheo1) [inline app/src/main/java/io/github/polysmee/settings/FriendsActivity.java:66]: General comment for the entire class : don't forget to add javadoc on any method that isn't private. Here all methods are protected so they should have javadoc. Or you can make them private if they aren't used anywhere else

**Review by Morpheo1 — APPROVED** [review]
Appart from the small javadoc comment, everything seems good

**Review by MrZaiko — APPROVED** [review]
Good work!

**Pull request merged**